app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -29,7 +29,10 @@ def create
 
   # PATCH/PUT /api/v1/bookmarks/1
   def update
-    if @bookmark.update(bookmark_params)
+    @bookmark.assign_attributes bookmark_params
+    @bookmark.tags = update_tags
```

Comment:
Whats the difference between `upsert_tags` and `update_tags`?

Reply:
`upsert` merges the existing tags with the new parameter provided tags and could probably be renamed better, update just turns the array of string into an array of `Tag` objects. I'm not too happy about this still, and might go with the autosave magic with `accepts_nested_attributes_for` stuff

Reply:
I'm also not super happy about the whole create method in this controller with the upsert_params bit